tests/test_analysis.py

```diff
@@ -420,50 +420,68 @@ def test_mix_scenarios():
     }
 
 
-def test_scenario_switcher(seed):
+def test_improved_healthsystem_and_care_seeking_scenario_switcher(seed):
     """Check the `ScenarioSwitcher` module can update parameter values in a manner similar to them being changed
-    directly after registration in the simulation (as would be done by the Scenario class)."""
+    directly after registration in the simulation (as would be done by the Scenario class).
+    And check that these parameters can be changed mid-way through the simulation."""
 
     sim = Simulation(start_date=Date(2010, 1, 1), seed=seed)
     sim.register(*(
-        fullmodel(resourcefilepath=resourcefilepath) + [ScenarioSwitcher(resourcefilepath=resourcefilepath)]
+        fullmodel(resourcefilepath=resourcefilepath) +
+        [ImprovedHealthSystemAndCareSeekingScenarioSwitcher(resourcefilepath=resourcefilepath)]
```

Comment:
This could be included in full model?

Reply:
Good spot again - see above my response. I think I'm happy to leave it as is.